lib/openhab/dsl/rules/triggers/changed.rb

```diff
@@ -124,7 +124,7 @@ def create_item_changed_trigger(item, from, to)
         #  second element is a Hash configuring trigger
         #
         def create_group_changed_trigger(group, from, to)
-          config = { 'groupName' => group.group.name }
+          config = { 'groupName' => group.name }
```

Comment:
Does this really work? Where does it get the name from?

Reply:
> Does this really work? Where does it get the name from?

From 
```
          item_missing_delegate { @group_item }
```

I've added a test scenario to check for this. It's not a comprehensive test. We'd need to add more tests to it I think.

Reply:
But if `members` return a `MemberOfGroup` object, isn't that what gets passed to the method here?

Reply:
> But if `members` return a `MembersOfGroup` object, isn't that what gets passed to the method here instead of the `GroupItem` object?

You are correct! I've made some adjustments now

Reply:
That would work, but is basically the same as the old implementation except that it uses a full class instead of a struct. And to iterate through the members you would have to write `Group.members.members.each`

Reply:
Also Group.each() should work because of this:
```
          item_missing_delegate { members }
```

Reply:
Not sure if it works anyway, created a small test script:
```ruby
require 'openhab'

class MembersOfGroup < Array
  def initialize(name, members)
    @group = name
    super(members)
  end
end

def meth(*items)
  items.flatten.each { |arg| logger.info(arg) }
end

group = MembersOfGroup.new(Switches, Switches.members.to_a)

meth(group)
```
Where Swithches is a Group with two members, Switch1 and Switch2.
Log output:
```
11:51:22.913 [INFO ] [jsr223.jruby.Object                  ] - Switch1 (Type=SwitchItem, State=ON, Label=Switch Number One, Category=, Groups=[Switches])
11:51:22.914 [INFO ] [jsr223.jruby.Object                  ] - Switch2 (Type=SwitchItem, State=ON, Label=Switch Number Two, Category=, Groups=[Switches])
```
Since the `changed` method calls `flatten` on the items, it does this recursively, which means that `each` would operate on every item individually, even if they are nested in a sub-array, which they are in practice since `MembersOfGroup.is_a?(Array)` is true.

Reply:
My guess is that `Array#flatten` works something like:
```ruby
def flatten
  flattened = []
  each do |obj|
    if obj.respond_to?(:flatten) 
      flattened.concat(obj.flatten)
    else
      flattened << obj
    end
  end
end
```
Which means we have to make sure that the object passed to the trigger methods either doesn't respond to `flatten` or override it to return self (which would cause problems if someone would want to use it in a rule)